### PR TITLE
[DISCUSSION] centralizing api calls

### PR DESCRIPTION
**Pull request by khamui**
@phillimorland @fuzzylogic2000 @Rineee @sabinammm 
today i felt like testing around to unify how we make api calls from the frontend.
In a4 we have something like that, but i think using the context API is more elegant.

it can -maybe- solve the issue described here: #4107

What you guys think?

Note: it is just a draft, not all api request methods are implemented yet.

### CODE REVIEW

File: meinberlin/apps/budgeting/assets/VoteButton.jsx [hunk 1]
```diff
@@ -1,64 +1,64 @@
 import React from 'react'
 import django from 'django'
-import { updateItem } from '../../contrib/assets/helpers.js'
+import { useApi } from '../../contrib/assets/ApiProvider'
 
-export default class VoteButton extends React.Component {
-  async addVote () {
+export const VoteButton = props => {
```
Comment (khamui) [inline meinberlin/apps/budgeting/assets/VoteButton.jsx:5]: it looks like a lot of changes, actually just converting from class notation to function notation.

**Comment by phillimorland** [comment]
@khamui would you be ok to go through the pros and cons of this over what we currently do after the next stand up where we are all there? just so we're all on the same page, and we can then discuss

**Comment by khamui** [comment]
> @khamui would you be ok to go through the pros and cons of this over what we currently do after the next stand up where we are all there? just so we're all on the same page, and we can then discuss

absolutely! lets do that next time we are all there.

**Comment by phillimorland** [comment]
closing for now but have written issue to record conclusion

**Pull request closed without merging**